packages/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/test/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsPluginTest.kt

```diff
@@ -0,0 +1,190 @@
+package expo.modules.plugin
+
+import com.google.common.truth.Truth
+import expo.modules.plugin.configuration.ExpoAutolinkingConfig
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.GradleRunner
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import java.io.File
+
+class ExpoAutolinkingSettingsPluginTest {
+  @JvmField
+  @Rule
+  var testProjectDir: TemporaryFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    testProjectDir.root.removeRecursively()
+    testProjectDir.root.createProject()
+  }
+
+  @Test
+  fun `applies settings plugin`() {
+    val result = executeGradleRun()
+    Truth.assertThat(result.output).contains("BUILD SUCCESSFUL")
+  }
+
+  @Test
+  fun `injects expo gradle extension`() {
+    val result = executeGradleRun(":app:gradleExpoExtension")
```

Comment:
nit: not sure whether run the e2e tests using junit is a good idea. another option is that you wrote e2e test using js at: https://github.com/expo/expo/tree/main/packages/expo-modules-autolinking/e2e/__tests__. maybe people would be more familiar with jest.

Reply:
The primary benefit of using JUnit with the Gradle test SDK is that we do not need to worry about providing the correct version of the settings plugin. So I think the setup is much simpler when using junit.